pandas/io/formats/csvs.py

```diff
@@ -62,7 +62,7 @@ def __init__(
         # Extract compression mode as given, if dict
         compression, self.compression_args = get_compression_method(compression)
 
-        self.path_or_buf, _, _, _ = get_filepath_or_buffer(
+        self.path_or_buf, _, _, self.should_close = get_filepath_or_buffer(
```

Comment:
@TomAugspurger - you pointer here was the exact problem https://github.com/pandas-dev/pandas/issues/32486#issuecomment-595992013